OSS CI Export Work

dsabeti opened the pull request.

Commit:
Add test.bash

with configure_db option

Signed-off-by: David Sabeti <sabetid@vmware.com>

Commit:
fix gofmt

Commit:
remove test.bash and use CI's default implementation

Commit:
Add test.bash for components that need to be ran in serial

dsabeti committed:
Stick with testing best practices

This commit contains an assortment of small fixes, some of which have turned out to be important:
- Randomizing test fixture names
- Adding temp files under a a temp directory like /tmp/test-01 instead of directly under /tmp
- Using non-deprecated functions

Signed-off-by: Amin Jamali <ajamali@vmware.com>

Commit:
Remove gosub submodule

Commit:
Move sync-package-specs to wg-app-platform-ci

Commit:
Remove unneeded scripts

Copied from wg-app-platform-runtime-ci examples/repo-scripts-dir/with-db

Commit:
Remove unneeded opsfiles

Commit:
Remove dependency on .envrc

- Reorgnanize scripts so that it's clear which scripts are ran
inside ofg docker containers and which are ran in the local workstation
- ./scripts/test-in-docker-locally.bash -> Helper to run Everything
- ./scripts/create-docker-container.bash -> Create a docker container to
  test and run things manually

Commit:
Update README

jrussett commented on scripts/create-docker-container.bash:
```diff
@@ -0,0 +1,52 @@
+echo $ARGS
+
+docker pull "${IMAGE}"
+docker rm -f "$REPO_NAME-docker-container"
```
By tying the name to the repo only (`"$REPO_NAME-docker-container"`) , one cannot use this script to test the repo with many different DB's at once. 

### Possible solution
Maybe we should use the DB in some capacity? Define the name as `${REPO_NAME}-${DB}-docker-container`?

For example:
- `silk-release-mysql-5.7-docker-container`
- `silk-release-postgres-docker-container`

jrussett commented on scripts/test-in-docker-locally.bash:
```diff
@@ -0,0 +1,18 @@
+DB="${DB}" "${THIS_FILE_DIR}/create-docker-container.bash" -d
+
+docker exec $REPO_NAME-docker-container '/repo/scripts/docker/tests-templates.bash'
+docker exec $REPO_NAME-docker-container '/repo/scripts/docker/test.bash' "$@"
```
I'm seeing transient errors when I was testing this.

`scripts/test-in-docker-locally.bash` appears to consistently fail in the silk teardown suite (`src/code.cloudfoundry.org/silk/teardown`).

But then manual invocation of `/repo/scripts/docker/test.bash` from within a running container works just fine.

dsabeti committed:
Add timestamp to docker container name

dsabeti committed:
Move auto-remove logic to a separate script

dsabeti committed:
Revert "Add timestamp to docker container name"

This reverts commit a2caffd829bf11b636e031d39d6ad2f2e1e80678.

dsabeti committed:
create-docker-container.bash takes an argument for what command to run

- Defaults to /bin/bash if not provided
- User can still provide docker flags using DOCKER_ARGS environment variable

dsabeti committed:
Randomize container names to keep them unique

dsabeti committed:
Pass test scripts to create-docker-container.bash

Previously, we were creating a container, backgrounding it, and then execing individual commands inside that container.
This change passes those commands directly to the container at start up time.
This allows us to use randomized container names

dsabeti committed:
Use --rm flag to remove old containers

jrussett reviewed (approved):
Looks good to me!
